### PR TITLE
[IOTDB-5334] Optimize metric related code and log

### PR DESCRIPTION
**Pull request by SpriCoder**
1. Optimize metric related code according to IDEA warning.
2. Optimize metric related log.

### CODE REVIEW

File: metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricManager.java [hunk 1]
```diff
@@ -436,15 +443,15 @@ protected boolean stop() {
 
   protected abstract boolean stopFramework();
 
-  private boolean isValid(MetricLevel metricLevel, String name, String... tags) {
+  private boolean inValid(MetricLevel metricLevel, String name, String... tags) {
```
Comment (OneSizeFitsQuorum) [inline metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricManager.java:446]: maybe `invalid` would be better...

File: metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricService.java [hunk 2]
```diff
@@ -367,15 +369,15 @@ public AbstractMetricManager getMetricManager() {
...
+  /** Bind metrics and store metric set. */
   public void addMetricSet(IMetricSet metricSet) {
     if (!metricSets.contains(metricSet)) {
       metricSet.bindTo(this);
       metricSets.add(metricSet);
     }
   }
 
-  /** Remove metrics */
+  /** Remove metrics.. */
```
Comment (OneSizeFitsQuorum) [inline metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricService.java:380]:
```suggestion
  /** Remove metrics. */
```

File: metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/reporter/DropwizardJmxReporter.java [hunk 3]
```diff
@@ -47,18 +47,26 @@ public boolean start() {
...
   public boolean stop() {
-    if (jmxReporter != null) {
-      jmxReporter.stop();
-      jmxReporter = null;
+    try {
+      if (jmxReporter != null) {
+        jmxReporter.stop();
+        jmxReporter = null;
+      }
+    } catch (Exception e) {
```
Comment (OneSizeFitsQuorum) [inline metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/reporter/DropwizardJmxReporter.java:65]: Which exception will be catched here?

File: metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java [hunk 4]
```diff
@@ -28,6 +28,12 @@
 public class DropwizardMetricNameTool {
   private static final String TAG_SEPARATOR = ".";
 
+  /**
+   * Transform flat string and metric type to metricInfo
```
Comment (OneSizeFitsQuorum) [inline metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java:32]:
```suggestion
   * Transform flat string and metric type to metricInfo.
```

File: metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java [hunk 5]
```diff
@@ -48,6 +54,11 @@ public static MetricInfo transformFromString(MetricType metricType, String flatS
     return metricInfo;
   }
 
+  /**
+   * Transform metricInfo to flat string
```
Comment (OneSizeFitsQuorum) [inline metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java:58]:
```suggestion
   * Transform metricInfo to flat string.
```

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricManager.java (hunk 1 above)
Comment (SpriCoder) [reply metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricManager.java:446]: get~

File: metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricService.java (hunk 2 above)
Comment (SpriCoder) [reply metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricService.java:380]: remove~

File: metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java (hunk 5 above)
Comment (SpriCoder) [reply metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java:58]: get

File: metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java (hunk 4 above)
Comment (SpriCoder) [reply metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/DropwizardMetricNameTool.java:32]: get

File: metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/reporter/DropwizardJmxReporter.java (hunk 3 above)
Comment (SpriCoder) [reply metrics/dropwizard-metrics/src/main/java/org/apache/iotdb/metrics/dropwizard/reporter/DropwizardJmxReporter.java:65]: To catch unexpected exception throw by jmxReporter stop method

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**